### PR TITLE
Move checkdataset to crab client repo fix 5233

### PR DESCRIPTION
**Pull request by belforte**
also introduce `commandUsedInsideCrab()` function in `ClientUtilities` so that we can decide if/when to use `print()` in a command for extra-verbosity of "progress report in same line" which only applies when using `crab` command, not the python Client API

### CODE REVIEW

**Comment by belforte** [comment]
HOLD
this PR is broken somewhere
```
something has gone wrong in one of last commits. Since now
```
belforte@lxplus808/TC3> crab status 
Traceback (most recent call last):
  File "/afs/cern.ch/user/b/belforte/WORK/CRAB3/CRABClient/bin/crab.py", line 22, in <module>
    from CRABClient.ClientUtilities import getAvailCommands, initLoggers, setConsoleLogLevelVar, StopExecution, flushMemoryLogger, LOGFORMATTER
  File "/afs/cern.ch/user/b/belforte/WORK/CRAB3/CRABClient/src/python/CRABClient/ClientUtilities.py", line 32, in <module>
    from CRABClient.UserUtilities import getUsernameFromCRIC
  File "/afs/cern.ch/user/b/belforte/WORK/CRAB3/CRABClient/src/python/CRABClient/UserUtilities.py", line 17, in <module>
    from CRABClient.ClientUtilities import DBSURLS, LOGLEVEL_MUTE, colors
ImportError: cannot import name 'DBSURLS' from partially initialized module 'CRABClient.ClientUtilities' (most likely due to a circular import) (/afs/cern.ch/user/b/belforte/WORK/CRAB3/CRABClient/src/python/CRABClient/ClientUtilities.py)
belforte@lxplus808/TC3> 
```
```

**Comment by belforte** [comment]
I think cleanest solution is to more `getUsernameFromCRIC` from `UserUtilities` to `ClientUtilities` since users are satisfied with the "technology independent" `getUsername`.
I guess when I switched from SiteDB to CRIC I did not dare to change too much.
The error was triggered when moving the import from "inside the code" to top of file, but this way it is better.

**Comment by belforte** [comment]
ready for review. Unfortunately we need to commit to a branch in main repo in order to test with Jenkins.

**Comment by belforte** [comment]
Jenkins OK for CMSSW 13 and 7 
* https://github.com/dmwm/CRABServer/issues/8021
* https://github.com/dmwm/CRABServer/issues/8024

for other release need to fix Jenkins first https://github.com/dmwm/CRABServer/issues/8025

**Comment by belforte** [comment]
@novicecpp the problems with Jenkins have been understood and this file now "compliles" in python2 too. Ready for review. Let me know if you prefer to call you out of this, then I merge and "since it is a new funcitonality, we can surely find and fix bugs later on"

**Review by novicecpp — COMMENTED** [review]
There are some f-string lines in `checkfile.py`. 
Do you plan to fix later in https://github.com/dmwm/CRABClient/pull/5262?

**Comment by belforte** [comment]
yes it is done in there. I did in two PR to avoid having to wait for this to be merged before doing the other.

**Comment by belforte** [comment]
a bit of a mess :-( My fault

File: src/python/CRABClient/Commands/checkfile.py [hunk 1]
```diff
@@ -93,7 +94,8 @@ def __call__(self):
             self.logger.info(f"\nCheck Adler32 checksum ({self.fileToCheck['adler32']}) for each disk replica")
         for rse in rseWithSizeOK:
             pfn = diskReplicas[rse]
-            print(f"verify checksum for replica at {rse}. ", end="", flush=True)
+            if commandUsedInsideCrab():
+                print(f"verify checksum for replica at {rse}. ", end="", flush=True)
```
Comment (novicecpp) [inline src/python/CRABClient/Commands/checkfile.py:98]:
```suggestion
            if commandUsedInsideCrab():
                print("verify checksum for replica at %s. " % rse, end="", flush=True)
```

**Review by novicecpp — APPROVED** [review]
Fix my suggestion a bit (because those are new lines here), and you are good to go :)

**Pull request merged**